alien4cloud-core/src/main/java/org/alien4cloud/alm/deployment/configuration/services/MatchingSubstitutionService.java

```diff
@@ -149,14 +147,13 @@ public void onCopyConfiguration(OnDeploymentConfigCopyEvent onDeploymentConfigCo
 
     private List<ITopologyModifier> getMatchingFlow() {
         List<ITopologyModifier> modifierList = flowExecutor.getDefaultFlowModifiers();
+        NodeMatchingModifier nodeMatchingModifier = (NodeMatchingModifier) modifierList.stream().filter(modifier -> modifier instanceof NodeMatchingModifier)
+                .findFirst().orElseThrow(() -> new IllegalArgumentException(
+                        "Unexpected exception in deployment flow to update node substitution; unable to find the master node matching modifier to inject selection action modifier."));
 
-        for (int i = 0; i < modifierList.size(); i++) {
```

Comment:
You may need to keep that actually as right now you cleanup matching sub-modifiers but not modifiers that are called after the matching global modifier.
Previous logic was removing all modifiers after that as they are not required.